src/modules/fs/api.c

```diff
@@ -20,22 +26,119 @@ int writeToFile(const char *filename, const char *data) {
     return 1;
 }
 
-char *readFromFile(const char *filename) {
-    FILE* f = fopen(filename, "rb");
+char *readFromFile(const char *filename)
+{
+    FILE *f = fopen(filename, "rb");
     char *str = malloc(1);
     int len = 1;
 
-    if (f == NULL) {
+    if (f == NULL)
+    {
         return NULL;
     }
 
-    while (!feof(f)) {
+    while (!feof(f))
+    {
         str = realloc(str, len + 1);
-        str[len - 1] = (char) fgetc(f);
+        str[len - 1] = (char)fgetc(f);
         len++;
     }
     str[len - 2] = '\0';
 
     fclose(f);
     return str;
 }
+
+int deleteFile(const char *filename)
+{
+    char command[256];
+    sprintf(command, "rm %s", filename);
+
+    if (system(command) == 0)
+    {
+        return 1;
+    }
+    else
+    {
+        return 0;
+    }
+}
+
+int deleteDirectory(const char *dirname)
+{
+    char command[256];
+    sprintf(command, "rm -r %s", dirname);
+
+    if (system(command) == 0)
+    {
+        return 1;
+    }
+    else
+    {
+        return 0;
+    }
+}
+
+int exists(const char *path)
+{
+    struct stat s;
+    int err = stat(path, &s);
+
+    if (-1 == err) {
+        return 0; // Does not exist
+    } else {
+        return 1; // Exists
+    }
+}
+
+int existsFile(const char *filename)
+{
+    struct stat s;
+    int err = stat(filename, &s);
+
+    if (-1 == err) {
+        return 0; // Does not exist
+    } else {
+        if (S_ISDIR(s.st_mode))
+        {
+            return 0; // Exists as a directory
+        }
+        else
+        {
+            return 1; // Exists as a file
+        }
+    }
+}
+
+int existsDirectory(const char *dirname)
+{
+    struct stat s;
+    int err = stat(dirname, &s);
+    if (-1 == err) {
+        return 0; // Does not exist
+    } else {
+        if (S_ISDIR(s.st_mode))
+        {
+            return 1; // Exists as a directory
+        }
+        else
+        {
+            return 0; // Exists as a file
+        }
+    }
+}
+
+int createDirectory(const char *dirname)
+{
+    char command[256];
+    sprintf(command, "mkdir %s", dirname);
```

Comment:
Don't call system commands, what if they aren't present? see man page for `mkdir(2)`